core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -2115,6 +2161,40 @@ object ZIOSpec extends ZIOBaseSpec {
         }
       }
     ),
+    suite("validateM")(
+      testM("returns all errors if never valid") {
+        val in  = List.fill(10)(0)
+        val res = ZIO.validateM[Any, Int, Int, Int](in)(a => ZIO.fail(a)).flip
+        assertM(res, equalTo(in))
+      },
+      testM("accumulate errors and ignore successes") {
+        val in  = List.range(0, 10)
+        val res = ZIO.validateM(in)(a => if (a % 2 == 0) ZIO.effect(a) else ZIO.fail(a))
```

Comment:
`ZIO.succeed` here and elsewhere. If you have issues with `CanFail` you can import `zio.CanFail.canFail` to silence them for the purpose of testing.